code/edgeidentity/src/test/java/com/adobe/marketing/mobile/edge/identity/IdentityStateTests.java

```diff
@@ -247,36 +253,29 @@ public void testBootupIfReady_whenIdentityDirectRegistered_onFirstBoot_waitsForI
 						new HashMap<String, String>() {
 							{
 								put("friendlyName", "Identity");
-								put("version", "1.2.2");
+								put("version", "2.0.0");
 							}
 						}
 					);
 				}
 			}
 		);
-		state.bootupIfReady(mockSharedStateCallback);
 
-		verify(mockSharedPreferenceEditor, never()).apply();
+		when(mockSharedStateCallback.getSharedState(IdentityConstants.SharedState.Hub.NAME, null))
+			.thenReturn(new SharedStateResult(SharedStateStatus.SET, hubSharedState));
+		when(mockSharedStateCallback.getSharedState(IdentityConstants.SharedState.IdentityDirect.NAME, null))
+			.thenReturn(null);
 
-		// verify
-		assertNull(state.getIdentityProperties().getECID());
-		assertEquals(0, setXDMSharedEventStateCalledTimes);
+		assertFalse(identityState.bootupIfReady(mockSharedStateCallback));
+		assertNull(identityState.getIdentityProperties().getECID());
+		verify(mockIdentityStorageManager, never()).savePropertiesToPersistence(identityState.getIdentityProperties());
+		verify(mockSharedStateCallback, times(0)).createXDMSharedState(any(), any());
 	}
 
 	@Test
-	public void testBootupIfReady_whenIdentityDirectRegistered_onFirstBoot_usesIdentityDirectECID() {
-		// setup
-		Mockito
-			.when(mockContext.getSharedPreferences(IdentityConstants.DataStoreKey.IDENTITY_DIRECT_DATASTORE_NAME, 0))
-			.thenReturn(mockSharedPreference);
-		Mockito
-			.when(mockSharedPreference.getString(IdentityConstants.DataStoreKey.IDENTITY_DIRECT_ECID_KEY, null))
-			.thenReturn(null);
-
-		IdentityState state = new IdentityState(new IdentityProperties());
-
-		// test
-		hubSharedState = new HashMap<>();
+	public void testBootUpIfReady_waitsForPendingIdentityDirectStateIfRegistered_stateIsNull() {
+		final IdentityState identityState = new IdentityState(mockIdentityStorageManager);
```

Comment:
nit: Like I mentioned above we could reword it like `..._stateDataIsNull`

Reply:
reword it to `waitsForPendingIdentityDirectStateToResolveIfRegistered` we can drop `stateIsNull` as it doesnt affect the outcome